docs(guides) add advanced entry usage example with different filetypes (#3158)
webpack/webpack.js.org: src/content/guides/entry-advanced.md

---
title: Advanced entry
sort: 24
contributors:
  - EugeneHlushko
---

## Multiple file types per entry

> It is possible to provide different types of files when using arrays for [entry](/configuration/entry-context/#entry) value(s) to achieve separate bundles for CSS and JavaScript (and other) files in applicataions that are not using `import` for styles in JavaScript (pre Single Page Applications or different reasons).

Reviewer: ```suggestion
It is possible to provide different types of files when using an array of values for [entry](/configuration/entry-context/#entry) to achieve separate bundles for CSS and JavaScript (and other) files in applicataions that are not using `import` for styles in JavaScript (pre Single Page Applications or different reasons).
```

> It is possible to provide different types of files when using arrays for [entry](/configuration/entry-context/#entry) value(s) to achieve separate bundles for CSS and JavaScript (and other) files in applicataions that are not using `import` for styles in JavaScript (pre Single Page Applications or different reasons).

Reviewer: ```suggestion
It is possible to provide different types of files when using arrays for [entry](/configuration/entry-context/#entry) value(s) to achieve separate bundles for CSS and JavaScript (and other) files in applications that are not using `import` for styles in JavaScript (pre Single Page Applications or different reasons).
```

> It is possible to provide different types of files when using an array of values for [entry](/configuration/entry-context/#entry) to achieve separate bundles for CSS and JavaScript (and other) files in applicataions that are not using `import` for styles in JavaScript (pre Single Page Applications or different reasons).

Author: ```suggestion
It is possible to provide different types of files when using an array of values for [entry](/configuration/entry-context/#entry) to achieve separate bundles for CSS and JavaScript (and other) files in applications that are not using `import` for styles in JavaScript (pre Single Page Applications or different reasons).
```

Let's make an example. We have a PHP application with two page types: home and account. The home page has different layout and non-sharable JavaScript with the rest of the application (account page). We want to output `home.js` and `home.css` from our application files for the home page and `account.js` and `account.css` for account page.

__home.js__

```javascript
console.log('home page type');
```

__home.scss__

```scss
// home page individual styles
```

__account.js__

```javascript

> console.log('accout page type');

Reviewer: ```suggestion
console.log('account page type');
```

```

__account.scss__

```scss

> // accout page individual styles

Reviewer: ```suggestion
// account page individual styles
```

```

> We will use [MiniCssExtractPlugin](/plugins/mini-css-extract-plugin/) in 'production' mode for css as a best practice.

Reviewer: ```suggestion
We will use [MiniCssExtractPlugin](/plugins/mini-css-extract-plugin/) in `production` mode for css as a best practice.
```
